snuba/snuba_migrations/spans_v2/0001_spans_v2.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.schemas import (
+    UUID,
+    Bool,
+    Column,
+    DateTime,
+    DateTime64,
+    Float,
+    Map,
+    String,
+    UInt,
+)
+
+storage_set_name = StorageSetKey.SPANS_V2
+local_table_name = "spans_v2_local"
+dist_table_name = "spans_v2_dist"
+
+columns: List[Column[Modifiers]] = (
+    [
+        Column("organization_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("trace_id", UUID()),
+        Column("span_id", UInt(64)),
+        Column("parent_span_id", UInt(64, Modifiers(codecs=["ZSTD(1)"]))),
+        Column("segment_id", UInt(64, Modifiers(codecs=["ZSTD(1)"]))),
+        Column("segment_name", String(Modifiers(default="''", codecs=["ZSTD(1)"]))),
+        Column("is_segment", Bool()),
+        Column(
+            "_sort_timestamp", DateTime(Modifiers(codecs=["DoubleDelta", "ZSTD(1)"]))
+        ),
+        Column(
+            "start_timestamp",
+            DateTime64(6, modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
+        ),
+        Column(
+            "end_timestamp",
+            DateTime64(6, modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
+        ),
+        Column(
+            "duration_ms",
+            UInt(32, modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
+        ),
+        Column("exclusive_time_ms", Float(64, modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column(
+            "retention_days",
+            UInt(16, modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
+        ),
+        Column("name", String(modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column("sampling_factor", Float(64, modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column("sampling_weight", Float(64, modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column("sign", UInt(8, modifiers=Modifiers(codecs=["DoubleDelta(1)"]))),
+    ]
+    + [
+        Column(
+            f"attr_str_{i}",
+            Map(String(), String(), modifiers=Modifiers(codecs=["ZSTD(1)"])),
+        )
+        for i in range(50)
+    ]
+    + [
+        Column(
+            f"attr_num_{i}",
+            Map(String(), Float(64), modifiers=Modifiers(codecs=["ZSTD(1)"])),
+        )
+        for i in range(50)
+    ]
+    + [
+        Column(
+            f"attr_bool_{i}",
+            Map(String(), Bool(), modifiers=Modifiers(codecs=["ZSTD(1)"])),
+        )
+        for i in range(10)
+    ]
+)
+
+index_create_ops: List[SqlOperation] = (
```

Comment:
I have a suspicion we're going to want some kind of index on project_id and segment_name. Not sure if we want to add something for that now, or wait and add them later.

Reply:
i'm skeptical a DSI on project ID would do much, maybe hold off for now